src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+#include "Look/WaypointLook.hpp"
+#include "Look/NavigatorLook.hpp"
+#include "Look/IconLook.hpp"
+#include "Look/MapLook.hpp"
+#include "Look/TaskLook.hpp"
+#include "Math/Constants.hpp"
+#include "ProgressBarRenderer.hpp"
+#include "Task/ProtectedTaskManager.hpp"
+#include "Engine/Task/TaskManager.hpp"
+#include "Engine/Task/Ordered/OrderedTask.hpp"
+#include "Engine/Task/Ordered/Points/OrderedTaskPoint.hpp"
+#include "Components.hpp"
+#include "Engine/Task/Unordered/AlternateList.hpp"
+#include "Formatter/LocalTimeFormatter.hpp"
+#include "Formatter/TimeFormatter.hpp"
+#include "ui/canvas/Font.hpp"
+#include "ui/dim/Rect.hpp"
+
+
+// standard
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+
+void
+NavigatorRenderer::DrawFrame(Canvas &canvas, const PixelRect &rc, const NavigatorLook &look) {
+  const auto top_left = rc.GetTopLeft();
+  const int rc_height = rc.GetHeight();
+  const int rc_width = rc.GetWidth();
+  
+  BulkPixelPoint polyline_frame[11];
+  polyline_frame[0].x = top_left.x + 5 / 20.0 * rc_height; //
+  polyline_frame[0].y = top_left.y; //
+  polyline_frame[1].x = top_left.x + rc_width - 4 / 20.0 * rc_height; //
+  polyline_frame[1].y = polyline_frame[0].y; //
+  polyline_frame[2].x = top_left.x + rc_width - 2 / 20.0 * rc_height; //
+  polyline_frame[2].y = top_left.y + 1 / 20.0 *rc_height;; //
+  polyline_frame[3].x = top_left.x + rc_width; //
+  polyline_frame[3].y = top_left.y + 10 / 20.0 * rc_height; //
+  polyline_frame[4].x = polyline_frame[2].x; //
+  polyline_frame[4].y = top_left.y + 19 / 20.0 *rc_height; //
+  polyline_frame[5].x = polyline_frame[1].x; //
+  polyline_frame[5].y = top_left.y + rc_height; //
+  polyline_frame[6].x = polyline_frame[0].x; //
+  polyline_frame[6].y = polyline_frame[5].y; //
+  polyline_frame[7].x = top_left.x + 2 / 20.0 * rc_height ; //
+  polyline_frame[7].y = polyline_frame[4].y; //
+  polyline_frame[8].x = top_left.x; //
+  polyline_frame[8].y = polyline_frame[3].y;//
+  polyline_frame[9].x = polyline_frame[7].x; //
+  polyline_frame[9].y = polyline_frame[2].y; //
+  polyline_frame[10].x = polyline_frame[0].x; //
+  polyline_frame[10].y = polyline_frame[0].y; //
+  
+  canvas.Select(look.background_pen);
+  canvas.Select(look.background_brush);
+  canvas.DrawPolygon( polyline_frame, 11);
+
+  canvas.Select(look.frame_pen);
+  canvas.Select(look.frame_brush);
+  canvas.DrawPolyline( polyline_frame, 11);
+}
+
+
+void 
+NavigatorRenderer::DrawText(Canvas &canvas, const Waypoint &wp_current, 
+                                const PixelRect &rc, const NavigatorLook &look) {
+
+  const auto &basic = CommonInterface::Basic();
+  const auto &calculated = CommonInterface::Calculated();
+  TCHAR current_speed[256] = "---";
+
+  if (basic.ground_speed_available)
+    FormatSpeed(current_speed, Units::ToSysUnit(basic.ground_speed, Unit::METER_PER_SECOND),
+              Unit::KILOMETER_PER_HOUR, true, true);
+  
+  auto current_speed_s = static_cast<std::string>(current_speed);
```

Comment:
What the .... you cast `TCHAR[]` to `std::string`... ?? Does that even compile?

Reply:
now, I'm re-coding the .... most beauty part of this development.... (!)
After reading again the part of the method 'NavigatorRenderer::DrawText', I must admit that it's a very 'funny' even strange code... but it did the job (for prototyping) and fortunately compile (!)

I found some other similar code and  I can get rid of the std::string and std::ostringstream.

for instance, current_speed_s stays a char array:
```
  // e_Speed_GPS
  TCHAR current_speed_s[10]{};
  FormatUserSpeed(basic.ground_speed, current_speed_s, true, 0);
```
moreover the function FormatUserSpeed() is very useful as it directly set the correct Unit set by the user.